src/features/Folding.ts

```diff
@@ -121,6 +124,9 @@ class LineNumberRange {
     ): LineNumberRange {
         this.startline = document.positionAt(start.startIndex).line;
         this.endline = document.positionAt(end.startIndex).line;
+        if (this.settings.codeFolding && this.settings.codeFolding.showLastLine) {
+            this.endline--;
```

Comment:
I assume `endline-1` is never going to be a bad value for VSCode. We might want to test some edge cases here.

Reply:
Acutally, in retrospect.  This is wrong place to do this.  Because, changing the endline too early means the `isValidRange` function will give out false positives (https://github.com/PowerShell/vscode-powershell/blob/7afec44453836b9b9b54b0505e7ba2857d9b0f84/src/features/Folding.ts#L153)

Instead it should replace https://github.com/PowerShell/vscode-powershell/blob/7afec44453836b9b9b54b0505e7ba2857d9b0f84/src/features/Folding.ts#L160-L162

``` typescript
    public toFoldingRange(settings:ISettings): vscode.FoldingRange {
        if (settings.codeFolding && settings.codeFolding.showLastLine) {
            return new vscode.FoldingRange(this.startline, this.endline - 1, this.rangeKind);
        } else {
            return new vscode.FoldingRange(this.startline, this.endline, this.rangeKind);
        }
    }
```

And then change the `provideFoldingRanges` method to;

https://github.com/PowerShell/vscode-powershell/blob/7afec44453836b9b9b54b0505e7ba2857d9b0f84/src/features/Folding.ts#L255-L261

``` typescript
        const settings = this.currentSettings();
        return foldableRegions
            // It's possible to have duplicate or overlapping ranges, that is, regions which have the same starting
            // line number as the previous region. Therefore only emit ranges which have a different starting line
            // than the previous range.
            .filter((item, index, src) => index === 0 || item.startline !== src[index - 1].startline)
            // Convert the internal representation into the VSCode expected type
            .map((item) => item.toFoldingRange(settings));
    }

    private currentSettings(): ISettings { Settings.load(); }
```

The private method `currentSettings()` is needed so we can mock it in the unit tests to flick the behaviour.